fetchRebalancerData.js

```diff
@@ -9,38 +10,52 @@ const contractAbi = require('./UniswapNFTPositionManagerAbi.json');
 
 const uniswapPositionManagerContractAddress = '0xC36442b4a4522E871399CD717aBDD847Ab11FE88';
 const contract = new web3.eth.Contract(contractAbi, uniswapPositionManagerContractAddress);
-const instaContractAddress = '0x575D9121127bEc262Bc920F465d60A9418A63744';
-let finalTransactions;
+let transactionData;
 
-const getEvents = async (fromBlock) => {
+const getEvents = async (dsaAccounts) => {
+    // store all transaction hashes obtained from IncreaseLiquidity event emits in a set for quick search later on
     const transactionHashes = new Set();
 
-    await contract.getPastEvents("IncreaseLiquidity",
-        { fromBlock: 25698992, toBlock: 25774738 },
-        (err, events) => {
-            for (let i = 0; i < events.length; i++) {
-                // console.log(events[i].transactionHash);
-                transactionHashes.add(events[i].transactionHash);
-            }
-            // console.log(events[0]);
-            // transactionHash = events[0].transactionHash;
-        });
-
-    events = await contract.getPastEvents("DecreaseLiquidity",
-        { fromBlock: 25698992, toBlock: 25774738 }
-    );
-
-    // console.log(events);
-    const instaTransactionHashes = events.map(e => {
-        return {
-            transactionHash: e.transactionHash,
-            tokenId: e.returnValues.tokenId,
-            liquidity: e.returnValues.liquidity
-        };
-    }).filter(transaction => transactionHashes.has(transaction.transactionHash));
-    // console.log(instaTransactionHashes);
-
-    finalTransactions = await Promise.all(instaTransactionHashes.map(async (transaction) => {
+    let latestBlock = await web3.eth.getBlockNumber(); // get the latest block number
+    let currentFirstBlock = 0;
+    let allValidTransactionHashes = [];
+
+    while(currentFirstBlock <= latestBlock){
+        let currentToBlock = Math.min(latestBlock, currentFirstBlock + 100000);
+
+        // get past events of type IncreaseLiquidity between currentFirstBlock and currentToBlock block numbers from Uniswap contract
+        await contract.getPastEvents("IncreaseLiquidity",
```

Comment:
I think using `for` loop would be a great alternative. 
You can do this: 
```js
  for (let i = 25657968; i < latest; i += 3000) {
```

fromBlock: 1
toBlock: i + 3000

This will save a lot of time.

Reply:
Okay